homeassistant/components/fan/dyson.py

```diff
@@ -15,42 +15,71 @@
+ATTR_NIGHT_MODE = 'night_mode'
+ATTR_AUTO_MODE = 'auto_mode'
+ATTR_ANGLE_LOW = 'angle_low'
+ATTR_ANGLE_HIGH = 'angle_high'
+ATTR_FLOW_DIRECTION_FRONT = 'flow_direction_front'
+ATTR_TIMER = 'timer'
 
 DEPENDENCIES = ['dyson']
 DYSON_FAN_DEVICES = 'dyson_fan_devices'
 
 SERVICE_SET_NIGHT_MODE = 'dyson_set_night_mode'
+SERVICE_SET_ANGLE = 'dyson_set_angle'
+SERVICE_SET_FLOW_DIRECTION_FRONT = 'dyson_flow_direction_front'
+SERVICE_SET_TIMER = 'dyson_set_timer'
 
 DYSON_SET_NIGHT_MODE_SCHEMA = vol.Schema({
     vol.Required(CONF_ENTITY_ID): cv.entity_id,
-    vol.Required(CONF_NIGHT_MODE): cv.boolean,
+    vol.Required(ATTR_NIGHT_MODE): cv.boolean,
+})
+
+DYSON_SET_ANGLE_SCHEMA = vol.Schema({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(ATTR_ANGLE_LOW): cv.positive_int,
+    vol.Optional(ATTR_ANGLE_HIGH): cv.positive_int
+})
+
+DYSON_SET_FLOW_DIRECTION_FRONT_SCHEMA = vol.Schema({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(ATTR_FLOW_DIRECTION_FRONT): cv.boolean
+})
+
+DYSON_SET_TIMER_SCHEMA = vol.Schema({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(ATTR_TIMER): cv.positive_int
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Dyson fan components."""
-    from libpurecoollink.dyson_pure_cool_link import DysonPureCoolLink
+    from libpurecool.dyson_pure_cool_link import DysonPureCoolLink
+    from libpurecool.dyson_pure_cool import DysonPureCool
 
     _LOGGER.debug("Creating new Dyson fans")
     if DYSON_FAN_DEVICES not in hass.data:
         hass.data[DYSON_FAN_DEVICES] = []
 
     # Get Dyson Devices from parent component
-    for device in [d for d in hass.data[DYSON_DEVICES] if
-                   isinstance(d, DysonPureCoolLink)]:
-        dyson_entity = DysonPureCoolLinkDevice(hass, device)
-        hass.data[DYSON_FAN_DEVICES].append(dyson_entity)
+    for device in hass.data[DYSON_DEVICES]:
+        if isinstance(device, DysonPureCool):
+            dyson_entity = DysonPureCoolDevice(hass, device)
+            hass.data[DYSON_FAN_DEVICES].append(dyson_entity)
+        elif isinstance(device, DysonPureCoolLink):
+            dyson_entity = DysonPureCoolLinkDevice(hass, device)
+            hass.data[DYSON_FAN_DEVICES].append(dyson_entity)
 
     add_entities(hass.data[DYSON_FAN_DEVICES])
 
     def service_handle(service):
         """Handle the Dyson services."""
         entity_id = service.data.get(CONF_ENTITY_ID)
-        night_mode = service.data.get(CONF_NIGHT_MODE)
+        night_mode = service.data.get(ATTR_NIGHT_MODE)
+        angle_low = service.data.get(ATTR_ANGLE_LOW)
+        angle_high = service.data.get(ATTR_ANGLE_HIGH)
+        flow_direction_front = service.data.get(ATTR_FLOW_DIRECTION_FRONT)
+        timer = service.data.get(ATTR_TIMER)
         fan_device = next([fan for fan in hass.data[DYSON_FAN_DEVICES] if
                            fan.entity_id == entity_id].__iter__(), None)
```

Comment:
Make this a generator expression instead by using parenthesis instead of brackets, and remove the `__iter__()` call.